internal/cli/capact/crd.go

```diff
@@ -0,0 +1,126 @@
+package capact
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+
+	goretry "github.com/avast/retry-go"
+	"github.com/pkg/errors"
+	"k8s.io/apiextensions-apiserver/pkg/apihelpers"
+	apiextensionv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextension "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	v1 "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset/typed/apiextensions/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/util/retry"
+)
+
+// LoadCRDDefinition loads the CRD from a given location. Supports local file or URL.
+func LoadCRDDefinition(location string) (*apiextensionv1.CustomResourceDefinition, error) {
```

Comment:
**Extended scope:** As discussed, please also handle applying CRD for a given Capact version. Currently we support only latest main CRD or from local path. Thank you!